core/domain/suggestion_services_test.py

```diff
@@ -1007,6 +1003,198 @@ def test_get_translation_suggestion_ids_with_exp_ids_with_empty_exp_list(
                 suggestion_services
                 .get_translation_suggestion_ids_with_exp_ids([])), 0)
 
+    def test_get_translation_suggestions_waiting_longest_for_review_per_lang(
+            self):
+        # Create a Hindi translation suggestion associated with exploration id
+        # target_id_1.
+        with self.swap(
+            exp_fetchers, 'get_exploration_by_id',
+            self.mock_get_exploration_by_id):
+            with self.swap(
+                exp_domain.Exploration, 'get_content_html',
+                self.MockExploration.get_content_html):
+                suggestion_services.create_suggestion(
+                    suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                    suggestion_models.TARGET_TYPE_EXPLORATION,
+                    self.target_id_1, 1, self.author_id_1,
+                    self.add_translation_change_dict, 'test description')
+        # Create a Hindi translation suggestion associated with exploration id
+        # target_id_2.
+        with self.swap(
+            exp_fetchers, 'get_exploration_by_id',
+            self.mock_get_exploration_by_id):
+            with self.swap(
+                exp_domain.Exploration, 'get_content_html',
+                self.MockExploration.get_content_html):
+                suggestion_services.create_suggestion(
+                    suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                    suggestion_models.TARGET_TYPE_EXPLORATION,
+                    self.target_id_2, 1, self.author_id_1,
+                    self.add_translation_change_dict, 'test description')
+        # Create a Hindi translation suggestion associated with exploration id
+        # target_id_3.
+        with self.swap(
+            exp_fetchers, 'get_exploration_by_id',
+            self.mock_get_exploration_by_id):
+            with self.swap(
+                exp_domain.Exploration, 'get_content_html',
+                self.MockExploration.get_content_html):
+                suggestion_services.create_suggestion(
+                    suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                    suggestion_models.TARGET_TYPE_EXPLORATION,
+                    self.target_id_3, 1, self.author_id_1,
+                    self.add_translation_change_dict, 'test description')
+
+        suggestions = (
+            suggestion_services
+            .get_translation_suggestions_waiting_longest_for_review_per_lang(
+                'hi'))
+
+        # Assert that the suggestions are in the order that they were created.
+        self.assertEqual(len(suggestions), 3)
+        self.assertLessEqual(suggestions[0].target_id, self.target_id_1)
+        self.assertLessEqual(suggestions[1].target_id, self.target_id_2)
+        self.assertLessEqual(suggestions[2].target_id, self.target_id_3)
```

Comment:
What does verifying the ordering of the target IDs achieve? Shouldn't we be comparing review dates to check order? My expectation is that the specifics of IDs are an implementation detail not relied upon for production behavior or testing purposes.

Reply:
Thanks for clarifying. The part that confused me was the assertLessEqual bit since that implies order, so the updated tests seem a lot clearer to me.